pkg/providers/instancetype/instancetype.go

```diff
@@ -131,6 +131,7 @@ func (p *DefaultProvider) List(ctx context.Context, kc *corev1beta1.KubeletConfi
 		blockDeviceMappingsHash,
 		aws.StringValue((*string)(nodeClass.Spec.InstanceStorePolicy)),
 		aws.StringValue(nodeClass.Spec.AMIFamily),
+		nodeClass.Name,
```

Comment:
I'm a bit worried with how this aligns with other changes we've made in the past that modifies our controller's memory usage + cache hits. Realistically, if every other detail is the same, and only the nodeClass name is different, we shouldn't need to maintain two cache values, especially since the data we're storing is pretty large.

Reply:
I'm not sure if adding this requirement into the cloudprovider.InstanceType is the right way to do this in the long run, especially as the `nodeClass` requirement is applicable for all instance types in the NodePool. I'd like to explore the idea of adding in a `karpenter.sh/nodeclass-name` label in the upstream code, so that we can better model this in the scheduling code. @jonathan-innis @engedaam thoughts?

Reply:
> aligns with other changes we've made in the past that modifies our controller's memory usage + cache hits

IMO, this is the strongest point. This will definitely increase our memory footprint for storing these instance types in a cache. Given this trade-off, I'd agree that the neutral code is probably the best way to attack this, though I'd recommend that we take a format more akin to `<nodeClassRef.group>/<nodeClassRef.kind>: <nodeClassRef.name>`